metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -294,6 +294,17 @@ class BuildServerConnection private (
     }
   }
 
+  def buildTargetRun(
+      params: RunParams,
+      cancelPromise: Promise[Unit],
+  ): Future[RunResult] = {
+    val completableFuture = register(server => server.buildTargetRun(params))
+    cancelPromise.future.foreach { _ =>
+      completableFuture.cancel(true)
+    }
```

Comment:
this has no effect. This methods is called from `DebugProvider`, but promise is never completed so cancelation logic will never be called which results in dead code.

Reply:
Actually, the promise was being cancelled in the DebugRunner, but it was under a wrong message. Should have been under DisconnectRequest, but was under TerminateRequest.

Should work correctly and I added a test for that.

I also realised that running a main method if multiple are present was not possible, that's fixed too.